tests/next_tests/integration_tests/feature_tests/ffront_tests/test_as_offset.py

```diff
@@ -0,0 +1,210 @@
+        cases.ConstInitializer(3)
+    )()
+    ref = a[5:]
+    cases.verify(
+        cartesian_case,
+        testee,
+        a,
+        offset_field,
+        out[2:],
+        inout=out[2:],
+        offset_provider={"Ioff": IDim},
+        ref=ref,
+    )
+
+
+@pytest.mark.parametrize(
+    "dims, offset, offset_dim, off_dim, other_dim",
+    [
+        ([IDim, KDim], Ioff, IDim, 0, 1),
+        ([IDim, JDim], Ioff, IDim, 0, 1),
+        ([IDim, KDim], Koff, KDim, 1, 0),
+    ],
+)
+@pytest.mark.uses_dynamic_offsets
+def test_offset_field_rand_connectivity(
+    cartesian_case, dims, offset, offset_dim, off_dim, other_dim
+):
+    @gtx.field_operator
+    def testee(
+        a: gtx.Field[[dims[0], dims[1]], int], offset_field: gtx.Field[[dims[0], dims[1]], int]
+    ) -> gtx.Field[[dims[0], dims[1]], int]:
+        return a(as_offset(offset, offset_field))
+
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
+    a = cases.allocate(cartesian_case, testee, "a").extend({offset_dim: (0, 3)})()
+    offset_field = cartesian_case.as_field(
+        [dims[0], dims[1]],
+        np.random.randint(
+            0,
+            4,
+            size=(cartesian_case.default_sizes[dims[0]], cartesian_case.default_sizes[dims[1]]),
+        ),
+    )
+    ref = np.zeros(offset_field.asnumpy().shape, dtype=int)
+
+    for j in range(len(offset_field.asnumpy()[off_dim])):
+        for i in range(len(offset_field.asnumpy())):
+            if offset_dim.kind == "vertical":
+                y = offset_field.asnumpy()[i][j] + np.indices(offset_field.shape)[off_dim][i][j]
+                x = np.indices(offset_field.shape)[other_dim][i][j]
+            else:
+                x = offset_field.asnumpy()[i][j] + np.indices(offset_field.shape)[off_dim][i][j]
+                y = np.indices(offset_field.shape)[other_dim][i][j]
+            ref[i][j] = a[x][y]
+
+    cases.verify(
+        cartesian_case,
+        testee,
+        a,
+        offset_field,
+        out=out,
+        offset_provider={"offset": offset_dim, "Ioff": IDim, "Koff": KDim},
+        ref=ref,
+        comparison=lambda out, ref: np.all(out == ref),
+    )
+
+
+@pytest.mark.uses_dynamic_offsets
+def test_offset_field_3d(cartesian_case):
+    @gtx.field_operator
+    def testee(a: cases.IJKField, offset_field: cases.IJKField) -> cases.IJKField:
+        return a(as_offset(Ioff, offset_field))
+
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
+    a = cases.allocate(cartesian_case, testee, "a").extend({IDim: (0, 2)})()
+    offset_field = cases.allocate(cartesian_case, testee, "offset_field").strategy(
+        cases.ConstInitializer(2)
+    )()
+
+    cases.verify(
+        cartesian_case,
+        testee,
+        a,
+        offset_field,
+        out=out,
+        offset_provider={"Ioff": IDim, "Koff": KDim},
+        ref=a[2:],
+        comparison=lambda out, ref: np.all(out == ref),
+    )
```

Comment:
I noticed we might be missing another test case: let's say we have a 3D field but the offset field is only 2D. I think the expected semantic is probably as if the offset field would be broadcasted first. This might be related to my comment about
> if self._ndarray.ndim > 1 and restricted_connectivity_domain == new_domain: